### PR TITLE
New spell: Summon Panzergeist

### PR DESCRIPTION
**Pull request by PleasingFungus**
L3 Earth/Summoning. Call forth the spirit of your armour, bringing forth
a slow-moving ally that hits harder the heavier your armour is. Casting
this in robes creates a 40 HP, 4 AC ally that hits twice for 4 damage.
In plate armour, you get a 40 HP, 20 AC ally that hits twice for 60 damage.
Wham, pow! (Summon limit is 1, as with CCF.)

The design goal here is to help provide some incentives for heavier armour
characters to delve into spellcasting. Historically, spells like Haste and
Phase Shift tempted players in this direction. Those spells had their own
problems, but the 'hybrid' skilling they encouraged was fun. This spell
will still be inappropriate for many characters, but if it brings a few
'melee' characters to dabble in spellcasting or 'caster' characters to
put on heavier armour, I think that'll be great.

Numbers are very much a first draft. I can easily imagine that the damage
should scale differently or that HP should vary by armour type, for example.

### CODE REVIEW

**Comment by ebering** [comment]
I vibe with the whole "make more hybrid/interesting spells" but I also worry about adding more summons, since it is the 3rd most populous spell school ( 2nd until the removal of demonology). That could be addressed by trimming or revising some of the other summons, though (looking at you Shadow Creatures).

**Comment by PleasingFungus** [comment]
I did just remove two of them :P

**Comment by RojjaCebolla** [comment]
"The spirit of your armor", FWIW, feels like a very gamey concept, not very evocative especially if you're not wearing an artifact.  Do you have any other ideas for the theming of it?  Maybe it summons an ancestor, who approves of the ornateness of your armor?  Maybe some weird gravitational effect? I just don't feel like that ringmail i found on d2 has much of a "spirit" in it.

**Comment by PleasingFungus** [comment]
> The sweat and toil of the armourer working leather and steel infuses a spirit into their creation, a unique spark within every cuirass, pauldron and vambrace. This spell draws those sparks out of the wearer's armour and brings them together, creating a Panzergeist, a protective spirit that fights alongside the caster...

A bit verbose, but does that seem more fun? Remember, this is a pre-industrial setting (such as it is a setting) - there's no assembly lines churning out +0 ring mails!

**Comment by randomMesh** [comment]
I like the idea but i don't like the name. The word "Panzer" is german and means either "tank" or "amour" while the association with tank is more common. It would destroy my picture of a medival, magical world when suddenly tanks are involved.

**Comment by randomMesh** [comment]
Thank you for renaming that spell. However, the correct spelling would be "Rüstungsgeist" (if you want to keep it).

**Comment by PleasingFungus** [comment]
I'm aware, thanks! I'm just very unconfident in DCSS's ability to handle umlauts.

**Comment by randomMesh** [comment]
> umlauts

I didn't mean the umlaut. I am well aware that ASCII is the way to go in DCSS. I meant the missing 'gs' in the name.
In the german language, when you combine two words, you add an [epenthesis](https://en.wikipedia.org/wiki/Epenthesis)  between these.

Rustung + Geist = Rustung**s**geist

You missed out either the 'g' of Rustung or Geist and the 's' for the epenthesis.

I don't want to nitpick here, it's just a suggestion.

**Review by ebering — COMMENTED** [review]
After some more poking and prodding I'm less skeptical of this, even with the large number of summon spells generally. Got some commentary on the duration, but implementation-wise this looks good and seems to work.

**Comment by Goratrix** [comment]
Just a suggestion, maybe change the name to something more fitting the fantasy game lore? As a german speaker, this 'steampunk'-ish name feels really out of place to me...

File: crawl-ref/source/spl-summoning.cc [hunk 1]
```diff
@@ -132,6 +132,53 @@ spret cast_call_canine_familiar(int pow, god_type god, bool fail)
...
+    int mitm_slot = get_mitm_slot(10);
+    if (mitm_slot == NON_ITEM)
+    {
+        canned_msg(MSG_NOTHING_HAPPENS);
+        return spret::abort;
+    }
+
+    fail_check();
+
+    const int dur = 2 + div_rand_round(pow, 25);
```
[the remark this answers is not in the document]
Comment (PleasingFungus) [reply crawl-ref/source/spl-summoning.cc:154]: Very good points! I've made the recommended changes, thanks :)

**Pull request merged**